CHANGELOG.md

```diff
@@ -1,6 +1,14 @@
 # TestEZ Changelog
 
 ## Unreleased Changes
+* Added `expect.extend` which allows projects to register their own, opinionated expectations that integrates into `expect`. ([#142](https://github.com/Roblox/testez/pull/142))
+  * Modeled after [jest's implementation](https://jestjs.io/docs/en/expect#expectextendmatchers).
+  * Matchers are functions that should return an object with with two keys, boolean `pass` and a string `message`
+  * Like `context`, matchers introduced via `expect.extend` will be present on all nodes below the node that introduces the matchers.
+  * Limitations:
+    * `expect.extend` cannot be called from within `describe` blocks
+    * Custom matcher names cannot overwrite pre-existing matchers, including default matchers and matchers introduces from previous `expect.extend` calls.
+    * Custom matcher names may not start with "_"
```

Comment:
What's the rationale for this?

Reply:
Originally this was to prevent overwriting private members of expect(), since currently we write to the expectation object's self fields directly, though realistically we can guard against this by using metatable to access from a scoped table instead...

Reply:
Address, no longer the case 👍